File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    # Gets a number between 1 and 8
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break

    # Iterate through each row
    for row in range(height):
        # Print spaces at front
        for i in range(height - row - 1):
            print(" ", end="")

        # Print pyramids
        print_pyramid(row + 1)
        print("  ", end="")
        print_pyramid(row + 1)

        # Insert new line
        print()


# Prints n hashes in row n
def print_pyramid(n):
    for i in range(n):
        print("#", end="")


if __name__ == "__main__":
    main()

File: pset1/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    for (int i = 0; i < height; i++)
    {
        //Prints left pyramid
        for (int j = 0; j < height - i - 1; j++)
        {
            printf(" ");
        }

        for (int k = 0; k < i + 1; k++)
        {
            printf("#");
        }

        //Prints two spaces in centre
        printf("  ");

        //Prints right pyramid
        for (int p = 0; p < i + 1; p++)
        {
            printf("#");
        }

        printf("\n");
    }
}